packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx

```diff
@@ -225,6 +193,44 @@ export const Datepicker: ComponentWithAs<'div', DatepickerProps> &
     overrideProps: overrideDatepickerCalendarProps,
   });
 
+  const openStateToBooleanKnob = (openState: OpenState): boolean => {
+    return openState === OpenState.Open;
+  };
+
+  const onInputClick = (): void => {
+    if (openState === OpenState.Closed) {
+      setOpenState(OpenState.Open);
+    } else if (openState === OpenState.Open || openState === OpenState.Closing) {
+      // Keep popup open in case we can only enter the date through calendar.
+      if (props.allowTextInput) {
+        setOpenState(OpenState.Closed);
+      } else {
+        setOpenState(OpenState.Open);
+      }
+    }
+  };
+
+  const onInputChange = (e, target: { value: string }) => {
+    const parsedDate = props.parseDate(target.value);
+
+    setFormattedDate(target.value);
+    if (parsedDate) {
+      if (isRestrictedDate(parsedDate, calendarOptions)) {
+        setError('The selected date is from the restricted range.');
```

Comment:
That should be a prop, so consumers can put the localized version of the error

Reply:
Good point, changed.